js/hashtag-validation.js

```diff
@@ -0,0 +1,96 @@
+import { debounce } from './util.js';
+
+const HASHTAG_MAX_LENGTH = 20;
+const HASHTAG_QUANTITY = 5;
+
+const validateHashtagQuantity = (hashtags) => {
+  return hashtags.length > HASHTAG_QUANTITY;
+}
+
+const validateHasgtagLength = (hashtags) => {
+  for (let i = 0; i < hashtags.length; i++) {
+    if (hashtags[i].length > HASHTAG_MAX_LENGTH) {
+      return true;
+    }
+  }
+  return false;
+}
+
+const isHashtagStartsWithHash = (hashtags) => {
+  for (let i = 0; i < hashtags.length; i++) {
+    if (hashtags[i].charAt(0) !== '#') {
+      return true;
+    }
+  }
+  return false;
+}
+
+const isHashtagUnique = (hashtags) => {
+  return new Set(hashtags).size !== hashtags.length;
+}
+
+const isEmptyHashtag = (hashtagInput) => {
+  return hashtagInput.value === '#' || hashtagInput.value.includes('# ');
+}
+
+const hasSpecialSymbols = (hashtags) => {
+  for (let i = 0; i < hashtags.length; i++) {
+    for (let j = 1; j < hashtags[i].length; j++) {
+      if (!/^[0-9A-Za-z]+$/.test(hashtags[i][j])) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+
+const validateHashtags = (hashtagInput) => {
+
+  const validateHashtag = () => {
+    const hashtags = hashtagInput.value.toLowerCase().split(' ').filter((item) => {
```

Comment:
можно сократить, оставить только те элементы, у которых непустой текст
```suggestion
    const hashtags = hashtagInput.value.toLowerCase().split(' ').filter((item) => !!item
```